MediaBrowser.Providers/Plugins/TheTvdb/TvdbSeriesProvider.cs

```diff
@@ -75,27 +58,98 @@ public async Task<MetadataResult<Series>> GetMetadata(SeriesInfo itemId, Cancell
                 QueriedById = true
             };
 
-            if (!IsValidSeries(itemId.ProviderIds))
+            if (!IsValidSeries(itemId))
             {
                 result.QueriedById = false;
                 await Identify(itemId).ConfigureAwait(false);
             }
 
             cancellationToken.ThrowIfCancellationRequested();
 
-            if (IsValidSeries(itemId.ProviderIds))
+            if (IsValidSeries(itemId))
             {
                 result.Item = new Series();
                 result.HasMetadata = true;
 
-                await FetchSeriesData(result, itemId.MetadataLanguage, itemId.ProviderIds, cancellationToken)
+                await FetchSeriesMetadata(result, itemId.MetadataLanguage, itemId.ProviderIds, cancellationToken)
                     .ConfigureAwait(false);
             }
 
             return result;
         }
 
-        private async Task FetchSeriesData(MetadataResult<Series> result, string metadataLanguage, Dictionary<string, string> seriesProviderIds, CancellationToken cancellationToken)
+        private async Task<IEnumerable<RemoteSearchResult>> FetchSeriesSearchResult(SeriesInfo seriesInfo, CancellationToken cancellationToken)
+        {
+            var tvdbId = seriesInfo.GetProviderId(MetadataProvider.Tvdb);
+            if (string.IsNullOrEmpty(tvdbId))
+            {
+                var imdbId = seriesInfo.GetProviderId(MetadataProvider.Imdb);
+                if (!string.IsNullOrEmpty(imdbId))
+                {
+                    tvdbId = await GetSeriesByRemoteId(
+                        imdbId,
+                        MetadataProvider.Imdb.ToString(),
+                        seriesInfo.MetadataLanguage,
+                        cancellationToken).ConfigureAwait(false);
+                }
+            }
+
+            if (string.IsNullOrEmpty(tvdbId))
+            {
+                var zap2ItId = seriesInfo.GetProviderId(MetadataProvider.Zap2It);
+                if (!string.IsNullOrEmpty(zap2ItId))
+                {
+                    tvdbId = await GetSeriesByRemoteId(
+                        zap2ItId,
+                        MetadataProvider.Zap2It.ToString(),
+                        seriesInfo.MetadataLanguage,
+                        cancellationToken).ConfigureAwait(false);
+                }
+            }
```

Comment:
Can't the 2nd if be moved inside the first?
```suggestion

            if (string.IsNullOrEmpty(tvdbId))
            {
                var zap2ItId = seriesInfo.GetProviderId(MetadataProvider.Zap2It);
                if (!string.IsNullOrEmpty(zap2ItId))
                {
                    tvdbId = await GetSeriesByRemoteId(
                        zap2ItId,
                        MetadataProvider.Zap2It.ToString(),
                        seriesInfo.MetadataLanguage,
                        cancellationToken).ConfigureAwait(false);
                }
            }
        }
```

Reply:
There's no benefit to it. I moved it outside the first if to reduce nesting.